lib/api/src/rc_update_replica_access_time.cpp

```diff
@@ -9,14 +9,16 @@
 
 auto rc_update_replica_access_time(RcComm* _comm, const char* _json_input, char** _output) -> int
 {
-    if (!_json_input) {
+    if (!_comm || !_json_input || !_output) {
         return SYS_INVALID_INPUT_PARAM;
```

Comment:
Consider `SYS_INTERNAL_NULL_INPUT_ERR` or `USER__NULL_INPUT_ERR` here since they're a little more specific.

Reply:
I really dislike those two error codes. Yes, they are more specific, but they also include words that are slightly misleading (e.g. INTERNAL, USER).

Given this is iRODS 5, I'm considering introducing a generically named error code for null and replacing all instances of the other error codes with it. Time is unlikely to be on my side though.

Will think about this a little more.

Reply:
If I introduce a new error code, I'll limit its use to the new code in this PR.

Reply:
Yeah, the current error code is used in many of the other APIs, but this feels like a good time to consider a better error code.

NULL values only apply to pointers, I think.

Here are some ideas:
- `NULL_POINTER_ERROR`
- `INVALID_POINTER`
- `INVALID_NULL_POINTER`
- `INVALID_INPUT_ARGUMENT`
- `INVALID_INPUT_ARGUMENT_NULL_POINTER`
- `ILLEGAL_INPUT_ARGUMENT`
- `POINTER_IS_NULL`
- `INPUT_POINTER_IS_NULL`
- `NULL_POINTER_ARGUMENT`
- `NULL_POINTER_INPUT_ARGUMENT`
- `NULL_POINTER_NOT_ALLOWED`
- `BAD_POINTER`
- `ILLEGAL_POINTER_ARGUMENT`

Are there others?

Reply:
I like `INVALID_INPUT_ARGUMENT_NULL_POINTER`.

Reply:
After some discussion, we've agreed to move forward with `INVALID_INPUT_ARGUMENT_NULL_POINTER`.

This also means any new error codes which can be more specific about the error must include the `INVALID_INPUT_ARGUMENT_` prefix.